app/services/definitions.py

```diff
@@ -0,0 +1,543 @@
+                await handle_acapy_call(
+                    logger=self.deps.logger,
+                    acapy_call=self.deps.aries_controller.schema.get_schema,
+                    schema_id=schema_id,
+                )
+                for schema_id in schemas_created_ids.schema_ids
+                if schema_id
+            ]
+
+            if schemas:
+                if len(schemas) > 1:
+                    raise CloudApiException(  # pylint: disable=W0707
+                        f"Multiple schemas with name {schema.name} and version {schema.version} exist."
+                        + f"These are: `{str(schemas_created_ids.schema_ids)}`.",
+                        409,
+                    )
+                self.deps.logger.debug("Using updated schema id with new DID")
+                _schema: SchemaGetResult = schemas[0]
+            else:
+                # if schema already exists, we should at least fetch 1, so this should never happen
+                raise CloudApiException(
+                    "Could not publish schema.", 500
+                )  # pylint: disable=W0707
+
+            # Schema exists with different attributes
+            if set(_schema.var_schema.attr_names) != set(schema.attribute_names):
+                raise CloudApiException(
+                    "Error creating schema: Schema already exists with different attribute names."
+                    + f"Given: `{str(set(_schema.var_schema.attr_names))}`. "
+                    f"Found: `{str(set(schema.attribute_names))}`.",
+                    409,
+                )  # pylint: disable=W0707
+
+            result = credential_schema_from_acapy(_schema.var_schema)
+            self.deps.logger.info(
+                "Schema already exists on ledger. Returning schema definition: `{}`.",
+                result,
+            )
+            return result
+
+
+class SchemaRegistrar:
+    def __init__(self, deps: ServiceDependencies):
+        self.deps = deps
+
+    async def register_schema(self, schema_id: str):
+        self.deps.logger.debug("Registering schema after successful publish to ledger")
+        try:
+            await register_schema(schema_id=schema_id)
+        except TrustRegistryException as error:
+            # If status_code is 405 it means the schema already exists in the trust registry
+            # That's okay, because we've achieved our intended result:
+            #   make sure the schema is registered in the trust registry
+            self.deps.logger.info(
+                "Caught TrustRegistryException when registering schema. "
+                "Got status code {} with message `{}`",
+                error.status_code,
+                error.detail,
+            )
+            if error.status_code == 405:
+                self.deps.logger.info(
+                    "Status code 405 indicates schema is already registered, so we can continue"
+                )
+            else:
+                raise error
+
```

Comment:
If we have multiple classes for our service, I think it's better for each class to be in its own module.

Then again, having a class SchemaRegistrar, for encapsulating one method, feels like we can just have naked method `register_schema`, without the class pattern, like most of our code! I actually prefer having classes instead of naked methods, but it only makes sense if there's a lot of complexity and class helps simplify things

For instance, here, `registrar = SchemaRegistrar(deps)` is just giving us access to this one method, which we could instead just call by passing controller and logger. Logger doesn't even have to be passed. It's only when we want to maintain the bound context. Since we are going from route to service, it's probably ok for service to just define it's own module's logger at the top, we can .bind whatever we want here